### PR TITLE
Add support for refreshing a resource indicator

### PR DESCRIPTION
**Pull request by danielcooper**
During the previous resource indicator work, I missed the refresh token case.

This adds support for a refresh and some tests.

### CODE REVIEW

**Review by mihail-stanchev-hopin — APPROVED** [review]
lgtm

File: lib/doorkeeper/oauth/refresh_token_request.rb [hunk 1]
```diff
@@ -10,9 +10,9 @@ class RefreshTokenRequest < BaseRequest
       validate :client,       error: :invalid_client
       validate :client_match, error: :invalid_grant
       validate :scope,        error: :invalid_scope
+      validate :resource_indicators, error: :invalid_target
```
Comment (pulpecja) [inline lib/doorkeeper/oauth/refresh_token_request.rb:13]: this is really small, but if all `error`s are in one line, maybe it's worth doing the same with a new one? or remove this completely :)

File: spec/lib/oauth/helpers/resource_indicators_checker_spec.rb [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    let(:subject) do
+      described_class.valid?(
+        double(resource_indicators: Doorkeeper::OAuth::ResourceIndicators.from_array(preapproved_indicators)),
+        Doorkeeper::OAuth::ResourceIndicators.from_array(requested_indicators),
+      )
+    end
+
+    context "when preapproved is a superset" do
+      let(:preapproved_indicators) do
+        ["http://example.com/1", "http://example.com/2"]
```
Comment (mihail-stanchev-hopin) [inline spec/lib/oauth/helpers/resource_indicators_checker_spec.rb:20]: Is it worth including a case for exact match and/or duplicates resource indicators?

**Review by pulpecja — APPROVED** [review]
I really like specs :)

File: spec/lib/oauth/helpers/resource_indicators_checker_spec.rb (hunk 2 above)
Comment (danielcooper) [reply spec/lib/oauth/helpers/resource_indicators_checker_spec.rb:20]: yeah don't see why not

**Pull request merged**